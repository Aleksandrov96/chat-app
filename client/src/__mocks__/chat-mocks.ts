export const currentChatroomMock = {
  _id: '63624569b610594d455aada2',
  participants: [
    {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    {
      _id: '6346b6fc037fb36c421c1694',
      email: 'pavlo@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
  ],
  createdAt: '2022-11-02T10:24:41.559Z',
  updatedAt: '2022-11-30T14:00:05.745Z',
  isTyping: false,
  latestMessage: {
    _id: '638761e5399dd8d9613f1cb2',
    sender: {
      _id: '6346b6fc037fb36c421c1694',
      email: 'pavlo@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'wrwr',
    chatroom: '63624569b610594d455aada2',
    isViewed: false,
    createdAt: '2022-11-30T14:00:05.608Z',
    updatedAt: '2022-11-30T14:00:05.781Z',
  },
};

export const currentChatMessagesMock = [
  {
    sender: {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'yo',
    chatroom: {
      _id: '63624569b610594d455aada2',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346b6fc037fb36c421c1694',
          email: 'pavlo@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-02T10:24:41.559Z',
      updatedAt: '2022-12-07T09:03:58.763Z',
      __v: 0,
      latestMessage: '639056fee2372570678adbea',
    },
    isViewed: true,
    _id: '63905ff9e2372570678add33',
    createdAt: '2022-12-07T09:45:17.386Z',
    updatedAt: '2022-12-07T09:45:17.386Z',
    __v: 0,
  },
  {
    sender: {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'hello',
    chatroom: {
      _id: '63624569b610594d455aada2',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346b6fc037fb36c421c1694',
          email: 'pavlo@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-02T10:24:41.559Z',
      updatedAt: '2022-12-07T09:03:58.763Z',
      __v: 0,
      latestMessage: '639056fee2372570678adbea',
    },
    isViewed: true,
    _id: '63905ff9e2372570678add35',
    createdAt: '2022-12-07T09:42:17.386Z',
    updatedAt: '2022-12-07T09:42:17.386Z',
    __v: 0,
  },
];

export const messageMock = {
  sender: {
    _id: '6346b85c487f713bee4a392f',
    email: 'aleksandrov@gmail.com',
    picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
  },
  content: 'test',
  chatroom: {
    _id: '63624569b610594d455aada2',
    participants: [
      {
        _id: '6346b85c487f713bee4a392f',
        email: 'aleksandrov@gmail.com',
        picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
      },
      {
        _id: '6346b6fc037fb36c421c1694',
        email: 'pavlo@gmail.com',
        picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
      },
    ],
    createdAt: '2022-11-02T10:24:41.559Z',
    updatedAt: '2022-12-07T09:03:58.763Z',
    __v: 0,
    latestMessage: '639056fee2372570678adbea',
  },
  isViewed: false,
  _id: '63905ff9e2372570678add33',
  createdAt: '2022-12-07T09:42:17.386Z',
  updatedAt: '2022-12-07T09:42:17.386Z',
  __v: 0,
};

export const chatsMock = [
  {
    _id: '63624569b610594d455aada2',
    participants: [
      {
        _id: '6346b85c487f713bee4a392f',
        email: 'aleksandrov@gmail.com',
        picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
      },
      {
        _id: '6346b6fc037fb36c421c1694',
        email: 'pavlo@gmail.com',
        picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
      },
    ],
    isTyping: false,
    createdAt: '2022-11-02T10:24:41.559Z',
    updatedAt: '2022-12-02T11:00:39.547Z',
    latestMessage: {
      _id: '6389dad710f13cccd8c01b1a',
      sender: {
        _id: '6346b6fc037fb36c421c1694',
        email: 'pavlo@gmail.com',
        picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
      },
      content: 'wer',
      chatroom: '63624569b610594d455aada2',
      isViewed: true,
      createdAt: '2022-12-02T11:00:39.379Z',
      updatedAt: '2022-12-02T11:00:39.601Z',
    },
  },
  {
    _id: '637c92d6b62a5be3173e82bb',
    participants: [
      {
        _id: '6346b85c487f713bee4a392f',
        email: 'aleksandrov@gmail.com',
        picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
      },
      {
        _id: '6346d39c4a8a0d337a45c998',
        email: 'chel2@gmail.com',
        picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
      },
    ],
    isTyping: false,
    createdAt: '2022-11-22T09:13:58.284Z',
    updatedAt: '2022-11-28T18:48:40.551Z',
    latestMessage: {
      _id: '638502889d2727fff49a2cd7',
      sender: {
        _id: '6346b6fc037fb36c421c1694',
        email: 'pavlo@gmail.com',
        picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
      },
      content: 'фів',
      chatroom: '637c92d6b62a5be3173e82bb',
      isViewed: true,
      createdAt: '2022-11-28T18:48:40.389Z',
      updatedAt: '2022-11-28T18:49:39.350Z',
    },
  },
];

export const userSearchResultsMock = [
  {
    email: 'chel2@gmail.com',
    _id: '6346d39c4a8a0d337a45c998',
    picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
  },
  {
    email: 'chel@gmail.com',
    _id: '6346b84f487f713bee4a3929',
    picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
  },
  {
    email: 'chuvak@gmail.com',
    _id: '6352baeeaa7a8891188ce227',
    picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
  },
];

export const chatWithoutLatestMessageMock = {
  _id: '63624569b610594d455aada2',
  participants: [
    {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    {
      _id: '6346b6fc037fb36c421c1694',
      email: 'pavlo@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
  ],
  isTyping: false,
  createdAt: '2022-11-02T10:24:41.559Z',
  updatedAt: '2022-12-02T11:00:39.547Z',
};

export const unviewedMessagesMock = [
  {
    _id: '638501459d2727fff49a2b19',
    sender: {
      _id: '6346d39c4a8a0d337a45c998',
      email: 'chel2@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'hello world!',
    chatroom: {
      _id: '637c92d6b62a5be3173e82bb',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346d39c4a8a0d337a45c998',
          email: 'chel2@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-22T09:13:58.284Z',
      updatedAt: '2022-11-28T18:48:40.551Z',
      latestMessage: '638502889d2727fff49a2cd7',
    },
    isViewed: false,
    createdAt: '2022-11-28T18:43:17.085Z',
    updatedAt: '2022-11-28T18:43:17.342Z',
  },
  {
    sender: {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'yo',
    chatroom: {
      _id: '63624569b610594d455aada2',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346b6fc037fb36c421c1694',
          email: 'pavlo@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-02T10:24:41.559Z',
      updatedAt: '2022-12-07T09:03:58.763Z',
      __v: 0,
      latestMessage: '639056fee2372570678adbea',
    },
    isViewed: false,
    _id: '63905ff9e2372570678add34',
    createdAt: '2022-12-07T09:42:17.386Z',
    updatedAt: '2022-12-07T09:42:17.386Z',
    __v: 0,
  },
  {
    sender: {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'hello!',
    chatroom: {
      _id: '63624569b610594d455aada2',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346b6fc037fb36c421c1694',
          email: 'pavlo@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-02T10:24:41.559Z',
      updatedAt: '2022-12-07T09:03:58.763Z',
      __v: 0,
      latestMessage: '639056fee2372570678adbea',
    },
    isViewed: false,
    _id: '63905ff9e2372570678add43',
    createdAt: '2022-12-07T09:42:17.386Z',
    updatedAt: '2022-12-07T09:42:17.386Z',
    __v: 0,
  },
];

export const currentChatMessagesWithUnviewedMock = [
  {
    sender: {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'yo',
    chatroom: {
      _id: '63624569b610594d455aada2',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346b6fc037fb36c421c1694',
          email: 'pavlo@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-02T10:24:41.559Z',
      updatedAt: '2022-12-07T09:03:58.763Z',
      __v: 0,
      latestMessage: '639056fee2372570678adbea',
    },
    isViewed: true,
    _id: '63905ff9e2372570678add33',
    createdAt: '2022-12-07T09:42:17.386Z',
    updatedAt: '2022-12-07T09:42:17.386Z',
    __v: 0,
  },
  {
    sender: {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'hello',
    chatroom: {
      _id: '63624569b610594d455aada2',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346b6fc037fb36c421c1694',
          email: 'pavlo@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-02T10:24:41.559Z',
      updatedAt: '2022-12-07T09:03:58.763Z',
      __v: 0,
      latestMessage: '639056fee2372570678adbea',
    },
    isViewed: true,
    _id: '63905ff9e2372570678add35',
    createdAt: '2022-12-07T09:42:17.386Z',
    updatedAt: '2022-12-07T09:42:17.386Z',
    __v: 0,
  },
  {
    sender: {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'test',
    chatroom: {
      _id: '63624569b610594d455aada2',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346b6fc037fb36c421c1694',
          email: 'pavlo@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-02T10:24:41.559Z',
      updatedAt: '2022-12-07T09:03:58.763Z',
      __v: 0,
      latestMessage: '639056fee2372570678adbea',
    },
    isViewed: false,
    _id: '63905ff9e2372570678addy4',
    createdAt: '2022-12-07T09:42:17.386Z',
    updatedAt: '2022-12-07T09:42:17.386Z',
    __v: 0,
  },
  {
    sender: {
      _id: '6346b85c487f713bee4a392f',
      email: 'aleksandrov@gmail.com',
      picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
    },
    content: 'hi there!',
    chatroom: {
      _id: '63624569b610594d455aada2',
      participants: [
        {
          _id: '6346b85c487f713bee4a392f',
          email: 'aleksandrov@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
        {
          _id: '6346b6fc037fb36c421c1694',
          email: 'pavlo@gmail.com',
          picture: 'https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png',
        },
      ],
      createdAt: '2022-11-02T10:24:41.559Z',
      updatedAt: '2022-12-07T09:03:58.763Z',
      __v: 0,
      latestMessage: '639056fee2372570678adbea',
    },
    isViewed: false,
    _id: '63905ff9e2372170678add43',
    createdAt: '2022-12-07T09:42:17.386Z',
    updatedAt: '2022-12-07T09:42:17.386Z',
    __v: 0,
  },
];
