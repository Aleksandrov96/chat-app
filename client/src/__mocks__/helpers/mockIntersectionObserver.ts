export function mockIntersectionObserver(): void {
  window.HTMLElement.prototype.scrollIntoView = () => {};
  const intersectionObserverMock = jest.fn();
  intersectionObserverMock.mockReturnValue({
    observe: () => null,
    unobserve: () => null,
    disconnect: () => null,
  });
  window.IntersectionObserver = intersectionObserverMock;
}
