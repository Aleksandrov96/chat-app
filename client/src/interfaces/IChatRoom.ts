import { IUser } from './IUser';

export interface IChatRoom {
  _id: string;
  participants: IUser[];
  latestMessage?: {
    _id: string;
    sender: IUser;
    content: string;
    chatroom: string;
    isViewed: boolean;
    createdAt: string;
    updatedAt: string;
  };
  isTyping: boolean;
  createdAt: string;
  updatedAt: string;
}
