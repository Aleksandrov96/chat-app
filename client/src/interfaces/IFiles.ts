export interface IFile {
  url: string,
  mimetype: string,
}
