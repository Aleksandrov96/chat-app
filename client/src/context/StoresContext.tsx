import React, { createContext } from 'react';
import AuthStore from 'stores/authStore';
import ChatStore from 'stores/chatStore';
import SocketStore from 'stores/socketStore';
import UIStore from 'stores/uiStore';
import { State } from 'interfaces/State';

const authStore = new AuthStore();
const chatStore = new ChatStore();
const socketStore = new SocketStore();
const uiStore = new UIStore();

export const StoresContext = createContext<Partial<State>>({
  authStore,
  chatStore,
  socketStore,
  uiStore,
});
