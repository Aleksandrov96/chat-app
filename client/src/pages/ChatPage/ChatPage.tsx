import React, { useContext } from 'react';
import { StoresContext } from 'context/StoresContext';
import Sidebar from 'components/Sidebar/Sidebar';
import Chat from 'components/Chat/Chat';
import ChatInfo from 'components/ChatInfo/ChatInfo';
import { ToasterContainer } from 'components/common/ToasterContainer/ToasterContainer';
import { observer } from 'mobx-react-lite';
import { IChatRoom } from 'interfaces/IChatRoom';
import { FormattedMessage } from 'react-intl';
import { useTheme } from 'hooks/useTheme';
import { ContextMenu } from './ContextMenu/ContextMenu';
import './chatPage.scss';

function ChatPage() {
  const { chatStore } = useContext(StoresContext);
  const { onThemeChange } = useTheme();

  return (
    <div className="chatPage">
      <Sidebar />
      {Object.entries(chatStore?.currentChatroom as IChatRoom).length > 0
        ? <Chat />
        : (
          <div className={onThemeChange('noChat')}>
            <p className="noChat__text">
              <FormattedMessage
                id="chatPage-no-chat-placeholder"
                values={{ placeholder: 'Select a chat to start messaging' }}
              />
            </p>
          </div>
        )}
      { chatStore?.showChatInfo && <ChatInfo /> }
      { chatStore?.showContextMenu && <ContextMenu /> }
      <ToasterContainer />
    </div>
  );
}

export default observer(ChatPage);
