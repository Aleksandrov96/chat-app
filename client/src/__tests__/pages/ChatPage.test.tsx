import React from 'react';
import ChatStore from 'stores/chatStore';
import { StoresContext } from 'context/StoresContext';
import ChatPage from 'pages/ChatPage/ChatPage';
import { currentChatroomMock } from '__mocks__/chat-mocks';
import { mockIntersectionObserver } from '__mocks__/helpers/mockIntersectionObserver';
import { render, screen, waitFor } from 'setupTests';
import { IChatRoom } from 'interfaces/IChatRoom';

beforeEach(() => {
  mockIntersectionObserver();
});

describe('ChatPage', () => {
  it('should render Chat component when chat is selected', async () => {
    const chatStore = new ChatStore();

    chatStore.currentChatroom = currentChatroomMock;

    const { asFragment } = render(
      <StoresContext.Provider value={{ chatStore }}>
        <ChatPage />
      </StoresContext.Provider>,
    );

    expect(screen.queryByTestId('chat')).toBeInTheDocument();
    expect(screen.queryByTestId('sidebar')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render dummy div when chat is not selected', async () => {
    const chatStore = new ChatStore();

    chatStore.currentChatroom = {} as IChatRoom;

    const { container, asFragment } = render(
      <StoresContext.Provider value={{ chatStore }}>
        <ChatPage />
      </StoresContext.Provider>,
    );

    expect(container.innerHTML).toMatch('Select a chat to start messaging');

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render custom context menu', async () => {
    const chatStore = new ChatStore();

    chatStore.showContextMenu = true;

    const { container } = render(
      <StoresContext.Provider value={{ chatStore }}>
        <ChatPage />
      </StoresContext.Provider>,
    );

    await waitFor(async () => {
      expect(container.innerHTML).toMatch('Delete chat');
    });
  });
});
