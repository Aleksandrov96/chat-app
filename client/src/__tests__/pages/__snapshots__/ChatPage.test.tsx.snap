// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatPage should render Chat component when chat is selected 1`] = `
<DocumentFragment>
  <div
    class="chatPage"
  >
    <div
      class="sidebar dark"
      data-testid="sidebar"
    >
      <div
        class="sidebarHeader dark"
      >
        <div
          class="user dark"
        >
          <img
            alt="User avatar"
            class="user__avatar dark"
          />
          <p
            class="user__name dark"
          />
        </div>
        <div
          class="wrapper"
        >
          <button
            aria-label="switch-theme-button"
            class="themeSwitcher dark"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-moon "
              data-icon="moon"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M223.5 32C100 32 0 132.3 0 256S100 480 223.5 480c60.6 0 115.5-24.2 155.8-63.4c5-4.9 6.3-12.5 3.1-18.7s-10.1-9.7-17-8.5c-9.8 1.7-19.8 2.6-30.1 2.6c-96.9 0-175.5-78.8-175.5-176c0-65.8 36-123.1 89.3-153.3c6.1-3.5 9.2-10.5 7.7-17.3s-7.3-11.9-14.3-12.5c-6.3-.5-12.6-.8-19-.8z"
                fill="currentColor"
              />
            </svg>
          </button>
          <select
            class="langSwitcher dark"
          >
            <option
              class="langSwitcher__option dark"
              value="en-US"
            >
              ðŸ‡¬ðŸ‡§
            </option>
            <option
              class="langSwitcher__option dark"
              value="uk"
            >
              ðŸ‡ºðŸ‡¦
            </option>
          </select>
          <button
            aria-label="signOut-button"
            class="signOut dark"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-right-from-bracket "
              data-icon="arrow-right-from-bracket"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M534.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L434.7 224 224 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l210.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128zM192 96c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0c-53 0-96 43-96 96l0 256c0 53 43 96 96 96l64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l64 0z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="searchBar"
      >
        <input
          class="search dark"
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
      <div
        class="chats dark"
        data-testid="chats"
      />
    </div>
    <div
      class="chat dark"
      data-testid="chat"
    >
      <div
        class="chatInfo dark"
      >
        <div
          class="wrapper"
        >
          <p
            class="chatInfo__name dark"
          >
            aleksandrov@gmail.com
          </p>
          <div
            class="typing"
          />
        </div>
        <button
          aria-label="show-chat-info-button"
          class="showInfo dark"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-ellipsis-vertical "
            data-icon="ellipsis-vertical"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 128 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M64 360c30.9 0 56 25.1 56 56s-25.1 56-56 56s-56-25.1-56-56s25.1-56 56-56zm0-160c30.9 0 56 25.1 56 56s-25.1 56-56 56s-56-25.1-56-56s25.1-56 56-56zM120 96c0 30.9-25.1 56-56 56S8 126.9 8 96S33.1 40 64 40s56 25.1 56 56z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="messages dark"
        data-testid="messages-container"
      >
        <div
          class="loader"
          data-testid="loader"
        >
          <div
            class="loadingio-spinner-rolling-qu81spmoyeb"
          >
            <div
              class="ldio-ncjl4jq3h27"
            >
              <div />
            </div>
          </div>
        </div>
        <div />
      </div>
      <div
        class="emojiPicker"
        data-testid="emoji-picker"
      />
      <div
        class="inputField"
      >
        <form
          class="messageForm"
          enctype="multipart/form-data"
          role="presentation"
        >
          <textarea
            class="messageForm__textarea dark"
            name="content"
            placeholder="Write a message..."
          />
          <div
            class="messageForm__icons"
          >
            <button
              aria-label="emoji-button"
              class="messageForm__icons-icon dark"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-face-smile "
                data-icon="face-smile"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM164.1 325.5C182 346.2 212.6 368 256 368s74-21.8 91.9-42.5c5.8-6.7 15.9-7.4 22.6-1.6s7.4 15.9 1.6 22.6C349.8 372.1 311.1 400 256 400s-93.8-27.9-116.1-53.5c-5.8-6.7-5.1-16.8 1.6-22.6s16.8-5.1 22.6 1.6zM208.4 208c0 17.7-14.3 32-32 32s-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32zm128 32c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <label
              class="messageForm__icons-icon dark"
              for="file"
            >
              <input
                accept=".jpg, .jpeg, .png, .mpeg, .mp3, .mp4, .wav, .gif"
                data-testid="file-input"
                id="file"
                name="file"
                style="display: none;"
                type="file"
              />
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-paperclip "
                data-icon="paperclip"
                data-prefix="fas"
                data-testid="file-upload-icon"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M396.2 83.8c-24.4-24.4-64-24.4-88.4 0l-184 184c-42.1 42.1-42.1 110.3 0 152.4s110.3 42.1 152.4 0l152-152c10.9-10.9 28.7-10.9 39.6 0s10.9 28.7 0 39.6l-152 152c-64 64-167.6 64-231.6 0s-64-167.6 0-231.6l184-184c46.3-46.3 121.3-46.3 167.6 0s46.3 121.3 0 167.6l-176 176c-28.6 28.6-75 28.6-103.6 0s-28.6-75 0-103.6l144-144c10.9-10.9 28.7-10.9 39.6 0s10.9 28.7 0 39.6l-144 144c-6.7 6.7-6.7 17.7 0 24.4s17.7 6.7 24.4 0l176-176c24.4-24.4 24.4-64 0-88.4z"
                  fill="currentColor"
                />
              </svg>
            </label>
            <button
              aria-label="message-submit-button"
              class="messageForm__icons-icon dark"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-paper-plane "
                data-icon="paper-plane"
                data-prefix="fas"
                data-testid="message-send-icon"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480V396.4c0-4 1.5-7.8 4.2-10.7L331.8 202.8c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8 17.7 316.6C7.1 311.3 .3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;

exports[`ChatPage should render dummy div when chat is not selected 1`] = `
<DocumentFragment>
  <div
    class="chatPage"
  >
    <div
      class="sidebar dark"
      data-testid="sidebar"
    >
      <div
        class="sidebarHeader dark"
      >
        <div
          class="user dark"
        >
          <img
            alt="User avatar"
            class="user__avatar dark"
          />
          <p
            class="user__name dark"
          />
        </div>
        <div
          class="wrapper"
        >
          <button
            aria-label="switch-theme-button"
            class="themeSwitcher dark"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-moon "
              data-icon="moon"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M223.5 32C100 32 0 132.3 0 256S100 480 223.5 480c60.6 0 115.5-24.2 155.8-63.4c5-4.9 6.3-12.5 3.1-18.7s-10.1-9.7-17-8.5c-9.8 1.7-19.8 2.6-30.1 2.6c-96.9 0-175.5-78.8-175.5-176c0-65.8 36-123.1 89.3-153.3c6.1-3.5 9.2-10.5 7.7-17.3s-7.3-11.9-14.3-12.5c-6.3-.5-12.6-.8-19-.8z"
                fill="currentColor"
              />
            </svg>
          </button>
          <select
            class="langSwitcher dark"
          >
            <option
              class="langSwitcher__option dark"
              value="en-US"
            >
              ðŸ‡¬ðŸ‡§
            </option>
            <option
              class="langSwitcher__option dark"
              value="uk"
            >
              ðŸ‡ºðŸ‡¦
            </option>
          </select>
          <button
            aria-label="signOut-button"
            class="signOut dark"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-right-from-bracket "
              data-icon="arrow-right-from-bracket"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M534.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L434.7 224 224 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l210.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128zM192 96c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0c-53 0-96 43-96 96l0 256c0 53 43 96 96 96l64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l64 0z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="searchBar"
      >
        <input
          class="search dark"
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
      <div
        class="chats dark"
        data-testid="chats"
      />
    </div>
    <div
      class="noChat dark"
    >
      <p
        class="noChat__text"
      >
        Select a chat to start messaging
      </p>
    </div>
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
